tools/rc/usResourceCompiler.cpp

```diff
@@ -0,0 +1,634 @@
+  std::unique_ptr<option::Option[]> buffer(new option::Option[stats.buffer_max]);
+  option::Parser parse(usage, argc, argv, options.get(), buffer.get());
+  
+  if (parse.error())
+  {
+    std::cerr << "Parsing command line arguments failed. " << std::endl;
+    return_code = EXIT_FAILURE;
+  }
+  
+  if (parse.nonOptionsCount())
+  {
+    std::clog << "unrecognized options ..." << std::endl;
+    for (int i = 0; i < parse.nonOptionsCount(); ++i)
+    {
+      std::cout << "\t" << parse.nonOption(i) << std::endl;
+    }
+    return_code = EXIT_FAILURE;
+  }
+  
+  option::Option* appendbinaryopt = options[APPENDBINARY];
+  if (appendbinaryopt && appendbinaryopt->count() > 1 )
+  {
+    std::cerr << "(--append-binary | -a) appear multiple times in the arguments. Check usage." << std::endl;
+    return_code = EXIT_FAILURE;
+  }
+  
+  option::Option* outfileopt = options[OUTFILE];
+  if (outfileopt && outfileopt->count() > 1 )
+  {
+    std::cerr << "(--out-file | -o) appear multiple times in the arguments. Check usage." << std::endl;
+    return_code = EXIT_FAILURE;
+  }
+  
+  if (!appendbinaryopt && !outfileopt)
+  {
+    std::cerr << "At least one of the options (--append-binary | --out-file) is required." << std::endl;
+    return_code = EXIT_FAILURE;
+  }
+  
+  if (argc == 0 || options[HELP] || return_code == EXIT_FAILURE)
+  {
+    option::printUsage(std::clog, usage);
+    return return_code;
+  }
+  
+  if (options[BUNDLENAME])
+  {
+    bundleName = options[BUNDLENAME].arg;
+  }
+  
+  if (!options[VERBOSE])
+  {
+    // if not in verbose mode, supress the clog stream
+    std::clog.setstate(std::ios_base::failbit);
+  }
+  
+  if (options[COMPRESSIONLEVEL])
+  {
+    char* endptr = 0;
+    compressionLevel = strtol(options[COMPRESSIONLEVEL].arg, &endptr, 10);
+  }
+  std::clog << "using compression level " << compressionLevel << std::endl;
+  
+  std::string outfile;
+  if (outfileopt)
+  {
+    outfile = outfileopt->arg;
+  }
+  else
+  {
+    outfile = us_tempfile();
+  }
+  
+  // ---------------------------------------------------------------------------------
+  //      OPEN OR CREATE ZIP FILE
+  // ---------------------------------------------------------------------------------
+  
+  mz_zip_archive writeArchive;
+  us_archived_names archivedNames;
+  us_archived_names archivedDirs;
+  memset(&writeArchive, 0, sizeof(writeArchive));
+  memset(&archivedNames, 0, sizeof archivedNames);
+  memset(&archivedDirs, 0, sizeof archivedDirs);
+  
+  
+  // ---------------------------------------------------------------------------------
+  //      ZIP ARCHIVE WRITING (temporary archive)
+  // ---------------------------------------------------------------------------------
+  
+  // Create a new zip archive which will be copied to zipfile later
+  std::clog << "Creating zip archive " << outfile << std::endl;
+  // clear the contents of a outfile if it exists
+  std::ofstream ofile(outfile, std::ofstream::trunc);
+  ofile.close();
```

Comment:
I am not sure how this is supposed to work in pure "append" mode (i.e. if I want to supply a bundle file path and an existing zip file that should be appended to the bundle shared library). The current code always truncates the output file (the bundle shared library in this case?)